examples/timeseries/timeseries_classification_from_scratch.py

```diff
@@ -0,0 +1,237 @@
+the number of classes beforehand.
+"""
+
+num_classes = len(np.unique(y_train))
+
+"""
+Now we shuffle the training set because we will be using the `validation_split` option
+later when training.
+"""
+
+idx = np.random.permutation(len(x_train))
+x_train = x_train[idx]
+y_train = y_train[idx]
+
+"""
+Standardize the labels to positive integers.
+"""
+
+y_train[y_train == -1] = 0
+y_test[y_test == -1] = 0
+
+"""
+## Build a model
+
+We build a Fully Convolutional Neural Network originally proposed in
+[this paper](https://arxiv.org/abs/1611.06455).
+The implementation is based on the TF2.0 version provided
+[here](https://github.com/hfawaz/dl-4-tsc/).
+The following hyperparameters (kernel_size, filters, the usage of BatchNorm) were found
+using the random search from [KerasTuner](https://github.com/keras-team/keras-tuner).
+
+"""
+
+
+def make_model(input_shape):
+    input_layer = keras.layers.Input(input_shape)
+
+    # This is the first 1D convolutional block.
+    # First a linear convolution is applied by sliding a certain number of filters
+    # on the input time series to transform it into a multivariate time series,
+    # whose number of dimensions is equal to the number of filters used here.
+    conv1 = keras.layers.Conv1D(filters=64, kernel_size=3, padding="same")(input_layer)
+    # Following the convolution, we apply a batchnormalization in order to
+    # help the network converge quickly while adding an implicit regularization.
+    conv1 = keras.layers.BatchNormalization()(conv1)
+    # Finally we add the relu activation function to inject some non-linearity
+    # to the convolutions' output.
+    conv1 = keras.layers.ReLU()(conv1)
+
+    conv2 = keras.layers.Conv1D(filters=64, kernel_size=3, padding="same")(conv1)
+    conv2 = keras.layers.BatchNormalization()(conv2)
+    conv2 = keras.layers.ReLU()(conv2)
+
+    conv3 = keras.layers.Conv1D(filters=64, kernel_size=3, padding="same")(conv2)
+    conv3 = keras.layers.BatchNormalization()(conv3)
+    conv3 = keras.layers.ReLU()(conv3)
+
+    # A global average pooling is used here.
+    # We average the time series over the whole time dimension.
+    # This would reduce drastically the number of parameters,
+    # while enabling the use of the class activation map method for interpretability.
+    # In fact the input time series with m dimensions is averaged
+    # resulting in a vector of m dimensions.
+    gap = keras.layers.GlobalAveragePooling1D()(conv3)
+
+    # The final softmax traditional classifier is used,
+    # with a number of neurons equal to the number of classes.
+    output_layer = keras.layers.Dense(num_classes, activation="softmax")(gap)
+
+    # We link the input and output layer by constructing the keras model.
+    return keras.models.Model(inputs=input_layer, outputs=output_layer)
+
+
+model = make_model(input_shape=x_train.shape[1:])
+keras.utils.plot_model(model, show_shapes=True)
+
+"""
+## Train the model
+
+"""
+
+epochs = 500
+batch_size = 128
```

Comment:
Because you only have 3601*0.8 training samples, you end up with 22 gradient update steps with a batch size of 128. This is very little, hence why you need to train for so many epochs. I would recommend using a smaller batch size, e.g. 32.